src/passes/externImporter.ts

```diff
@@ -16,6 +18,49 @@ import { NotSupportedYetError } from '../utils/errors';
 import * as pathLib from 'path';
 
 export class ExternImporter extends ASTMapper {
+  /*visitFunctionDefinition(node: FunctionDefinition, ast: AST): void {
+      node.vParameters.vParameters.forEach((parameter) => {
+        const typeNode = getNodeType(parameter, ast.compilerVersion);
+        const declarationSourceUnit = node.getClosestParentByType(SourceUnit);
+        const sourceUnit = node.getClosestParentByType(SourceUnit);
+
+        assert(sourceUnit !== undefined, 'Trying to import a definition into an unknown source unit');
+    if (declarationSourceUnit === undefined || sourceUnit === declarationSourceUnit) return; 
+        if (typeNode instanceof UserDefinedTypeName && 
+          typeNode.vReferencedDeclaration instanceof StructDefinition) {
+            ast.registerImport(node, formatPath(declarationSourceUnit.absolutePath), node.name);
+        }
+      });
+      return;
+    }*/
+  visitVariableDeclaration(node: VariableDeclaration, ast: AST): void {
+    const declaration = node.vType;
+    let declarationSourceUnit;
+    if (declaration instanceof UserDefinedTypeName) {
+      declarationSourceUnit = declaration.vReferencedDeclaration.getClosestParentByType(SourceUnit);
+    } else {
+      declarationSourceUnit = declaration?.getClosestParentByType(SourceUnit);
```

Comment:
If `node.vType` is not a UserDefinedType, there's no need to do anything other than recurse with a `this.commonVisit` and return. `node.vType` will always be in the same sourceUnit as `node`